ouroboros-consensus/src/Ouroboros/Storage/FS/Handle.hs

```diff
@@ -0,0 +1,61 @@
+{-# LANGUAGE LambdaCase #-}
+
+module Ouroboros.Storage.FS.Handle where
+
+import           Control.Concurrent.MVar
+import           Control.Exception
+import           System.IO.Error as IO
+
+-- import qualified Ouroboros.Storage.IO as F
+import           Ouroboros.Storage.FS.API.Types
+
+-- | File handlers for the IO instance for HasFS. 
+-- This is parametric on the os.
+--
+-- The file descriptor is provided by the HasFs and is used for Eq instance of the Handle.
+-- The 'FilePath' is used to improve error messages.
+-- The 'MVar' is used to implement 'close'.
+-- osHandle is Fd for unix and HANDLE for Windows.
+data HandleOS osHandle = Handle {
+      fid        :: Int
+    , handlePath :: FsPath
+    , filePath   :: FilePath
+    , handle     :: (MVar (Maybe osHandle))
```

Comment:
(Minor: Unnecessary brackets)

Reply:
Let's use the `handle` instead of `fid` for the equality (and then we don't need the counter anymore).